### PR TITLE
BUG: Fix lost precision when reading from Oracle Number column GH34988

### LINKED ISSUE #34988: BUG: Lost precision of NUMBER in read_sql_table().
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Problem description

When using `pd.read_sql_table()` with Oracle DB precision of data type `sqlalchemy.dialects.oracle.NUMBER` is lost.

#### Expected Output

Precision should not be lost as this datatype can be either integer or float.

#### The problem

Looks like the root cause for this is in `_harmonize_columns()`, particularly in `_get_dtype()` that is used there. The SQLAlchemy type `sqlalchemy.dialects.oracle.NUMBER` is based on both `sqlalchemy.types.Numeric` and `sqlalchemy.types.Integer`. As there is no support for SQLAlchemy Numeric type in `_get_dtype()` this Oracle type is being treated as integer and precision is dropped when using `astype()` in `_harmonize_columns()` itself.

### PR DESCRIPTION
**Pull request by andrewkisel**
Fixes: https://github.com/pandas-dev/pandas/issues/34988

- [x] closes #34988
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -1221,10 +1221,13 @@ def _get_dtype(self, sqltype):
             DateTime,
             Float,
             Integer,
+            Numeric,
         )
 
         if isinstance(sqltype, Float):
```
Comment (phofl) [inline pandas/io/sql.py:1227]: You could potentially move that into this isinstance check

**Review by phofl — COMMENTED** [review]
Can we create a test for this?

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a test which covers this

**Comment by mroeschke** [comment]
Thanks for the pull request, but appears it has gone stale. If you'd like to continue, please address the review, move the whatsnew note to v.1.5 and we can reopen

**Pull request closed without merging**